integration-test/helpers/harness.js

```diff
@@ -90,6 +90,23 @@ const setup = async (ops) => {
         spawnSync('rm', ['-rf', dataDir])
     }
 
+    if (ops.waitForConfiguration) {
+        // Wait for the extension to load its configuration files.
+        await bgPage.waitForFunction(
+            () => (
+                window.dbg &&
+                window.dbg.tds &&
+                !window.dbg.tds.isInstalling &&
+                window.dbg.tds.ClickToLoadConfig &&
+                window.dbg.tds.tds.trackers &&
+                window.dbg.tds.tds &&
+                window.dbg.tds.tds.domains &&
+                window.dbg.tds.tds.domains['facebook.com'] === 'Facebook, Inc.'
```

Comment:
is this something that we can generalize? will this entry always be in the configuration? I'm thinking that some tests might be mocking TDS and not having this entry.

Reply:
Uh, maybe? I'm open to suggestions. For now, I just put this behind the `waitForConfiguration` option so that we don't break the existing tests. At least that way there's a way to wait for all configuration to load for tests that want to do that.

Reply:
I've improved this logic now:
- I noticed that the `integration-tests/helpers/wait.js` module already took care of some aspects of background-page waiting, so I expanded on that and renamed the module to `backgroundWait.js`.
- Instead of hard-coding Facebook, I now check the first entity in the list.
- I made use of the fancy `?` syntantic sugar.
- I added some further checks for other aspects of the extension configuration.